rusoto/core/src/signature.rs

```diff
@@ -229,6 +229,97 @@ impl SignedRequest {
         self.params = params;
     }
 
+
+    /// http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html
+    pub fn generate_presigned_url(&mut self, creds: &AwsCredentials) -> String {
+        debug!("Presigning request URL");
+
+        self.sign(creds);
+        let hostname = match self.hostname {
+            Some(ref h) => h.to_string(),
+            None => build_hostname(&self.service, &self.region),
+        };
+
+        let current_time = now_utc();
+        let current_time_fmted = current_time.strftime("%Y%m%dT%H%M%SZ").unwrap();
+        let current_time_fmted = format!("{}", &current_time_fmted);
+        let current_date = current_time.strftime("%Y%m%d").unwrap();
+
+        self.remove_header("X-Amz-Content-Sha256");
+
+        self.remove_header("X-Amz-Date");
+        
+        self.remove_header("Content-Type");
+
+        if let Some(ref token) = *creds.token() {
+            self.remove_header("x-amz-security-token");
+            self.params.put("x-amz-security-token", encode_uri_strict(token));
+        }
+
+        self.remove_header("X-Amz-Algorithm");
+        self.params.put("X-Amz-Algorithm", "AWS4-HMAC-SHA256");
+
+        self.remove_header("X-Amz-Credential");
+        self.params.put("X-Amz-Credential", format!("{}/{}/{}/{}/aws4_request", &creds.aws_access_key_id(), &current_date, self.region.name(), self.service));
+
+        self.remove_header("X-Amz-Expires");
+        let expiration_time = match self.params.get("response-expires") {
```

Comment:
I think `response-expires` is wrong.
expires time should be pass as parameter.



https://github.com/boto/botocore/blob/939fbebcae0e0519f040063d61f7dc354cd09dd8/botocore/signers.py#L533-L534